ui/test/nuxeo-tooltip.test.js

```diff
@@ -0,0 +1,49 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html, fixture, isElementVisible, waitForAttrMutation } from '@nuxeo/testing-helpers';
+import '../widgets/nuxeo-tooltip.js';
+
+suite('nuxeo-tooltip', async () => {
+  test.only('Should not show a paper-tooltip if it has hidden atrribute set to true', async () => {
+    const tooltip = await fixture(
+      html`
+        <nuxeo-tooltip hidden="true">Hello</nuxeo-tooltip>
+      `,
+    );
+
+    tooltip.show();
+
+    const paperTooltip = document.body.querySelector('paper-tooltip');
+    expect(paperTooltip).to.be.null;
+  });
+
+  test.only('Should show a paper-tooltip if no hidden attribute is set', async () => {
+    const tooltip = await fixture(
+      html`
+        <nuxeo-tooltip>Hello</nuxeo-tooltip>
+      `,
+    );
+
+    tooltip.show();
+    const paperTooltip = document.body.querySelector('paper-tooltip');
+    const tooltipDiv = paperTooltip.shadowRoot.getElementById('tooltip');
+
+    await waitForAttrMutation(tooltipDiv, 'class');
+
+    expect(isElementVisible(paperTooltip)).to.be.true;
```

Comment:
I understand the rational here, but the border line is tricky.
With this approach we are testing the `paper-tooltip` internals. Which may not be the interesting case for our context.

What our component (`nuxeo-tooltip`) does it to add a `paper-tooltip` to `document.body`. Then it gets visible on the screen.
I believe this should be the scope of the test, as mentioned [here](https://github.com/nuxeo/nuxeo-elements/pull/233#discussion_r422098720).

Reply:
yes I understand :)
Unfortunately I couldn't find another way to test that the papertooltip show function was finished and we know for sure it will be visible. I did add tests to test that the paper tooltip is added with the correct content.